flask/app/analyses.py

```diff
@@ -244,6 +271,14 @@ def update_analysis(id: int):
         user_id = request.args.get("user")
     else:
         user_id = current_user.user_id
+        if request.json.get("analysis_state"):
+            if request.json.get("analysis_state") in [
```

Comment:
Don't need a nested if. `None in []` and `None in ["Foo"]` are both False and don't raise exceptions.